sdk/openai/Azure.AI.OpenAI/src/Custom/ChatCompletionsOptions.cs

```diff
@@ -15,28 +16,28 @@ namespace Azure.AI.OpenAI
     public partial class ChatCompletionsOptions
     {
         /// <inheritdoc cref="CompletionsOptions.ChoicesPerPrompt"/>
-        [CodeGenMember("N")]
         public int? ChoicesPerPrompt { get; set; }
 
         /// <inheritdoc cref="CompletionsOptions.FrequencyPenalty"/>
         public float? FrequencyPenalty { get; set; }
 
         /// <inheritdoc cref="CompletionsOptions.TokenSelectionBiases"/>
-        public IDictionary<int, int> TokenSelectionBiases { get; }
+        public IDictionary<int, int> TokenSelectionBiases
```

Comment:
This deserves its own dedicated discussion -- tokens are honestly a mess!

Conceptually, this is a map from positive integer token IDs (as computed via a "black box" algorithm that's purely SDK-external for the time being) to [-100, 100] integer bias values.

The major reason I don't want to skip deserializing these keys into a proper type is that string keys end up inherently confusable with the source string tokens themselves.

As a grounded example: a developer is likely to approach use of this with a purpose in mind like "I want to make the word 'rain' less likely to appear in my completions responses about weather forecasts."
- The correct path for this today is to use a library like `tiktoken` or ahead-of-time conversion via something like https://platform.openai.com/tokenizer to discover that "rain" has an ID of 3201, and then assign `TokenSelectionBiases[3201]` to a value like `-50`
- But without a fair degree of *a priori* knowledge, a developer seeing an `IDictionary<string, int>` is very likely to try setting `TokenSelectionBiases["rain"]` to `-50`, instead; if we're lucky, that'll at least give a descriptive error message -- but it'd be better to not rely on luck and guide towards the right path ahead of time

We're looking into the possibility of having better SDK representation of the token concept, including potentially having client-side calculation. Then this might become something like an `IDictionary<OpenAIStringToken, int>`, which is the kind of clarity we'd love to have. But in the interim, the underlying data type for the key is definitely an `int` and we want it to show up that way.

Looked at from another angle, the *only* reason we'd have this as a string is because it happens to be serialized on the wire that way--and that's just a coincidence of convenience! E.g. if `"logit_bias": { "3201": -50 }` were instead a slightly more verbose `"logit_bias": [ { "key": 3201, "value": -50 } ]`, we'd never consider `string` as the user-facing type.

Reply:
The generator doesn't support this today as there is no way to represent this in typespec we always assume Record<T> is a Dictionary<string, T>.

@johanste are there thoughts on whether typespec should allow you to specify the type of your key with some constraints on what would be valid here?